com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -504,14 +504,14 @@ internal void ServerDestroySpawnedSceneObjects()
                     // This **needs** to be here until we overhaul NetworkSceneManager due to dependencies
                     // that occur shortly after NetworkSceneManager invokes ServerDestroySpawnedSceneObjects
                     // within the NetworkSceneManager.SwitchScene method.
-                    SpawnedObjectsList.Remove(sobj);
+
                     if (NetworkManager.PrefabHandler != null && NetworkManager.PrefabHandler.ContainsHandler(sobj))
                     {
                         NetworkManager.PrefabHandler.HandleNetworkPrefabDestroy(sobj);
```

Comment:
This I am leaving up to the user to handle how they want to deal with scene transitions and prefab handler overrides.